### PR TITLE
Major version upgrade config

### PR DESCRIPTION
**Pull request by FxKu**
- [x] add fields to `OperatorConfiguration` CRD and their own category
- [x] copy major version upgrade configuration between CRD and internal config object
- [x] reflect in CRD validation
- [x] reflect in reference docs
- [x] update user and admin docs
- [x] update generated code

### CODE REVIEW

File: docs/user.md [hunk 1]
```diff
@@ -646,7 +646,13 @@ spec:
 
 ## In-place major version upgrade
 
-Starting with Spilo 13, operator supports in-place major version upgrade to a higher major version (e.g. from PG 10 to PG 12). To trigger the upgrade, simply increase the version in the manifest. It is your responsibility to test your applications against the new version before the upgrade; downgrading is not supported. The easiest way to do so is to try the upgrade on the cloned cluster first. For details of how Spilo does the upgrade [see here](https://github.com/zalando/spilo/pull/488), operator implementation is described [in the admin docs](administrator.md#minor-and-major-version-upgrade).
+Starting with Spilo 13, operator supports in-place major version upgrade to a
```
Comment (sdudoladov) [inline docs/user.md:649]: The upgrade capability also depends on the operator version, don;t forget to mention it here.

File: docs/administrator.md [hunk 2]
```diff
@@ -11,30 +11,43 @@ switchover (planned failover) of the master to the Pod with new minor version.
...
+Starting with Spilo 13, Postgres Operator can run an in-place major version
+upgrade which is much faster than cloning. First, you need to make sure, that
+the `PGVERSION` environment variable is set for the database pods. Since
+`v1.6.0` the related option `enable_pgversion_env_var` is enabled by default.
+
+In-place major version upgrades can be configured to be executed by the
+operator with the `major_version_upgrade_mode` option. By default it is set
+to `off` which means the cluster version will not change when increased in
+the manifest. Still, a rolling update would be triggered updating the
+`PGVERSION` variable. But `configure_spilo` will notice the version mismatch
```
Comment (sdudoladov) [inline docs/administrator.md:35]: worth mentioning `configure_spilo` is [a part of  Spilo](https://github.com/zalando/spilo/blob/849313c8e6be874c8f7193657fa78f25d7e2086a/postgres-appliance/scripts/configure_spilo.py#L934-L947) and not the operator.

File: docs/user.md (hunk 1 above)
Comment (Jan-M) [reply docs/user.md:649]: I actually think we dont need to specify either, this advertises to new users where we should assume starting with current release.

**Comment by FxKu** [comment]
:+1:

**Comment by Jan-M** [comment]
:+1:

**Comment by sdudoladov** [comment]
closed to re-trigger Zappr

**Pull request merged**